src/app.js

```diff
@@ -207,6 +222,61 @@ gpii.app.exit = function (that) {
     }
 };
 
+/**
+ * Extracts data for the user's preference sets (including the active preference
+ * set) from the message received when the user keys in.
+ * @param message {Object} The message sent when the user keys is (a JSON
+ * object).
+ * @return {Object} An object containing all preference sets and the active preference
+ * set.
+ */
+gpii.app.extractPreferencesData = function (message) {
+    var value = message.value || {},
+        preferences = value.preferences || {},
+        contexts = preferences.contexts,
+        activeContextName = value.activeContextName,
+        sets = [],
+        activeSet = null;
+
+    if (contexts) {
+        sets = fluid.hashToArray(contexts, "path");
+        activeSet = fluid.find_if(sets, function (preferenceSet) {
+            return preferenceSet.path === activeContextName;
+        });
+    }
+
+    return {
+        sets: sets,
+        activeSet: activeSet
+    };
+};
+
+/**
+ * Opens connection to the PCP Channel WebSocket
+ * @param that {Object} The app
+ */
+gpii.app.addCommunicationChannel = function (that) {
+    var socket = new ws("ws://localhost:8081/pcpChannel"); // eslint-disable-line new-cap
```

Comment:
Place this URL in some configuration for the component. Also, ensure that the socket is cleaned up when the component is destroyed - timed by its "onDestroy"

Reply:
These has been issued in these two commits: [socket config](https://github.com/danailbd/gpii-app/commit/74541630905260a8b54ac1bdfd5c84771ba3ab68#diff-bd9c9dcd314f2d7df52935b3a6a4d504L43) and [socket close](https://github.com/danailbd/gpii-app/commit/9e0ac546919db845c0dd88776d0831231b6dcc62#diff-bd9c9dcd314f2d7df52935b3a6a4d504R62)